File: v2/ruby/models/post_input.rb

class PostInput
  attr_accessor :created_at, :updated_at, :external_url, :external_url_source, :external_url_type, :image_path, :image_url, :image_height, :image_width, :logo_url, :square_logo_url, :needs_sanitize, :summary, :topic, :approval_status, :approval_transactions, :group_ids, :slug, :article_id, :comment, :newsletter_ids, :is_featured, :advisor_id, :is_published, :is_responsive, :is_proxy_needed, :is_mobile_proxy_needed, :proxy_url, :video, :pdf_attachment_url, :post_date, :display_date, :suitability_score, :video_id, :display_tag, :display_summary, :vestorly_url
  # :internal => :external
  def self.attribute_map
    {
      :created_at => :'created_at',
      :updated_at => :'updated_at',
      :external_url => :'external_url',
      :external_url_source => :'external_url_source',
      :external_url_type => :'external_url_type',
      :image_path => :'image_path',
      :image_url => :'image_url',
      :image_height => :'image_height',
      :image_width => :'image_width',
      :logo_url => :'logo_url',
      :square_logo_url => :'square_logo_url',
      :needs_sanitize => :'needs_sanitize',
      :summary => :'summary',
      :topic => :'topic',
      :approval_status => :'approval_status',
      :approval_transactions => :'approval_transactions',
      :group_ids => :'group_ids',
      :slug => :'slug',
      :article_id => :'article_id',
      :comment => :'comment',
      :newsletter_ids => :'newsletter_ids',
      :is_featured => :'is_featured',
      :advisor_id => :'advisor_id',
      :is_published => :'is_published',
      :is_responsive => :'is_responsive',
      :is_proxy_needed => :'is_proxy_needed',
      :is_mobile_proxy_needed => :'is_mobile_proxy_needed',
      :proxy_url => :'proxy_url',
      :video => :'video',
      :pdf_attachment_url => :'pdf_attachment_url',
      :post_date => :'post_date',
      :display_date => :'display_date',
      :suitability_score => :'suitability_score',
      :video_id => :'video_id',
      :display_tag => :'display_tag',
      :display_summary => :'display_summary',
      :vestorly_url => :'vestorly_url'
      
    }
  end

  def initialize(attributes = {})
    return if attributes.empty?
    # Morph attribute keys into undescored rubyish style
    
    if self.class.attribute_map[:"created_at"]
      @created_at = attributes["created_at"]
    end
    
    if self.class.attribute_map[:"updated_at"]
      @updated_at = attributes["updated_at"]
    end
    
    if self.class.attribute_map[:"external_url"]
      @external_url = attributes["external_url"]
    end
    
    if self.class.attribute_map[:"external_url_source"]
      @external_url_source = attributes["external_url_source"]
    end
    
    if self.class.attribute_map[:"external_url_type"]
      @external_url_type = attributes["external_url_type"]
    end
    
    if self.class.attribute_map[:"image_path"]
      @image_path = attributes["image_path"]
    end
    
    if self.class.attribute_map[:"image_url"]
      @image_url = attributes["image_url"]
    end
    
    if self.class.attribute_map[:"image_height"]
      @image_height = attributes["image_height"]
    end
    
    if self.class.attribute_map[:"image_width"]
      @image_width = attributes["image_width"]
    end
    
    if self.class.attribute_map[:"logo_url"]
      @logo_url = attributes["logo_url"]
    end
    
    if self.class.attribute_map[:"square_logo_url"]
      @square_logo_url = attributes["square_logo_url"]
    end
    
    if self.class.attribute_map[:"needs_sanitize"]
      @needs_sanitize = attributes["needs_sanitize"]
    end
    
    if self.class.attribute_map[:"summary"]
      @summary = attributes["summary"]
    end
    
    if self.class.attribute_map[:"topic"]
      @topic = attributes["topic"]
    end
    
    if self.class.attribute_map[:"approval_status"]
      @approval_status = attributes["approval_status"]
    end
    
    if self.class.attribute_map[:"approval_transactions"]
      @approval_transactions = attributes["approval_transactions"]
    end
    
    if self.class.attribute_map[:"group_ids"]
      if (value = attributes["group_ids"]).is_a?(Array)
        @group_ids = value
      end
    end
    
    if self.class.attribute_map[:"slug"]
      @slug = attributes["slug"]
    end
    
    if self.class.attribute_map[:"article_id"]
      @article_id = attributes["article_id"]
    end
    
    if self.class.attribute_map[:"comment"]
      @comment = attributes["comment"]
    end
    
    if self.class.attribute_map[:"newsletter_ids"]
      if (value = attributes["newsletter_ids"]).is_a?(Array)
        @newsletter_ids = value
      end
    end
    
    if self.class.attribute_map[:"is_featured"]
      @is_featured = attributes["is_featured"]
    end
    
    if self.class.attribute_map[:"advisor_id"]
      @advisor_id = attributes["advisor_id"]
    end
    
    if self.class.attribute_map[:"is_published"]
      @is_published = attributes["is_published"]
    end
    
    if self.class.attribute_map[:"is_responsive"]
      @is_responsive = attributes["is_responsive"]
    end
    
    if self.class.attribute_map[:"is_proxy_needed"]
      @is_proxy_needed = attributes["is_proxy_needed"]
    end
    
    if self.class.attribute_map[:"is_mobile_proxy_needed"]
      @is_mobile_proxy_needed = attributes["is_mobile_proxy_needed"]
    end
    
    if self.class.attribute_map[:"proxy_url"]
      @proxy_url = attributes["proxy_url"]
    end
    
    if self.class.attribute_map[:"video"]
      @video = attributes["video"]
    end
    
    if self.class.attribute_map[:"pdf_attachment_url"]
      @pdf_attachment_url = attributes["pdf_attachment_url"]
    end
    
    if self.class.attribute_map[:"post_date"]
      @post_date = attributes["post_date"]
    end
    
    if self.class.attribute_map[:"display_date"]
      @display_date = attributes["display_date"]
    end
    
    if self.class.attribute_map[:"suitability_score"]
      @suitability_score = attributes["suitability_score"]
    end
    
    if self.class.attribute_map[:"video_id"]
      @video_id = attributes["video_id"]
    end
    
    if self.class.attribute_map[:"display_tag"]
      @display_tag = attributes["display_tag"]
    end
    
    if self.class.attribute_map[:"display_summary"]
      @display_summary = attributes["display_summary"]
    end
    
    if self.class.attribute_map[:"vestorly_url"]
      @vestorly_url = attributes["vestorly_url"]
    end
    
  end

  def to_body
    body = {}
    self.class.attribute_map.each_pair do |key, value|
      body[value] = self.send(key) unless self.send(key).nil?
    end
    body
  end
end

File: v2/csharp/src/main/csharp/vestorly/Model/PostInput.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace vestorly.Model {
  [DataContract]
  public class PostInput {
    
    
    [DataMember(Name="created_at", EmitDefaultValue=false)]
    public string CreatedAt { get; set; }

    
    
    [DataMember(Name="updated_at", EmitDefaultValue=false)]
    public string UpdatedAt { get; set; }

    
    
    [DataMember(Name="external_url", EmitDefaultValue=false)]
    public string ExternalUrl { get; set; }

    
    
    [DataMember(Name="external_url_source", EmitDefaultValue=false)]
    public string ExternalUrlSource { get; set; }

    
    
    [DataMember(Name="external_url_type", EmitDefaultValue=false)]
    public string ExternalUrlType { get; set; }

    
    
    [DataMember(Name="image_path", EmitDefaultValue=false)]
    public string ImagePath { get; set; }

    
    
    [DataMember(Name="image_url", EmitDefaultValue=false)]
    public string ImageUrl { get; set; }

    
    
    [DataMember(Name="image_height", EmitDefaultValue=false)]
    public string ImageHeight { get; set; }

    
    
    [DataMember(Name="image_width", EmitDefaultValue=false)]
    public string ImageWidth { get; set; }

    
    
    [DataMember(Name="logo_url", EmitDefaultValue=false)]
    public string LogoUrl { get; set; }

    
    
    [DataMember(Name="square_logo_url", EmitDefaultValue=false)]
    public string SquareLogoUrl { get; set; }

    
    
    [DataMember(Name="needs_sanitize", EmitDefaultValue=false)]
    public string NeedsSanitize { get; set; }

    
    
    [DataMember(Name="summary", EmitDefaultValue=false)]
    public string Summary { get; set; }

    
    
    [DataMember(Name="topic", EmitDefaultValue=false)]
    public string Topic { get; set; }

    
    
    [DataMember(Name="approval_status", EmitDefaultValue=false)]
    public string ApprovalStatus { get; set; }

    
    
    [DataMember(Name="approval_transactions", EmitDefaultValue=false)]
    public List<string> ApprovalTransactions { get; set; }

    
    
    [DataMember(Name="group_ids", EmitDefaultValue=false)]
    public List<string> GroupIds { get; set; }

    
    
    [DataMember(Name="slug", EmitDefaultValue=false)]
    public string Slug { get; set; }

    
    
    [DataMember(Name="article_id", EmitDefaultValue=false)]
    public string ArticleId { get; set; }

    
    
    [DataMember(Name="comment", EmitDefaultValue=false)]
    public string Comment { get; set; }

    
    
    [DataMember(Name="newsletter_ids", EmitDefaultValue=false)]
    public List<string> NewsletterIds { get; set; }

    
    
    [DataMember(Name="is_featured", EmitDefaultValue=false)]
    public bool? IsFeatured { get; set; }

    
    
    [DataMember(Name="advisor_id", EmitDefaultValue=false)]
    public string AdvisorId { get; set; }

    
    
    [DataMember(Name="is_published", EmitDefaultValue=false)]
    public bool? IsPublished { get; set; }

    
    
    [DataMember(Name="is_responsive", EmitDefaultValue=false)]
    public bool? IsResponsive { get; set; }

    
    
    [DataMember(Name="is_proxy_needed", EmitDefaultValue=false)]
    public bool? IsProxyNeeded { get; set; }

    
    
    [DataMember(Name="is_mobile_proxy_needed", EmitDefaultValue=false)]
    public bool? IsMobileProxyNeeded { get; set; }

    
    
    [DataMember(Name="proxy_url", EmitDefaultValue=false)]
    public string ProxyUrl { get; set; }

    
    
    [DataMember(Name="video", EmitDefaultValue=false)]
    public string Video { get; set; }

    
    
    [DataMember(Name="pdf_attachment_url", EmitDefaultValue=false)]
    public string PdfAttachmentUrl { get; set; }

    
    
    [DataMember(Name="post_date", EmitDefaultValue=false)]
    public string PostDate { get; set; }

    
    
    [DataMember(Name="display_date", EmitDefaultValue=false)]
    public string DisplayDate { get; set; }

    
    
    [DataMember(Name="suitability_score", EmitDefaultValue=false)]
    public string SuitabilityScore { get; set; }

    
    
    [DataMember(Name="video_id", EmitDefaultValue=false)]
    public string VideoId { get; set; }

    
    
    [DataMember(Name="display_tag", EmitDefaultValue=false)]
    public string DisplayTag { get; set; }

    
    
    [DataMember(Name="display_summary", EmitDefaultValue=false)]
    public string DisplaySummary { get; set; }

    
    
    [DataMember(Name="vestorly_url", EmitDefaultValue=false)]
    public string VestorlyUrl { get; set; }

    
    
    [DataMember(Name="title", EmitDefaultValue=false)]
    public string Title { get; set; }

    
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class PostInput {\n");
      
      sb.Append("  CreatedAt: ").Append(CreatedAt).Append("\n");
      
      sb.Append("  UpdatedAt: ").Append(UpdatedAt).Append("\n");
      
      sb.Append("  ExternalUrl: ").Append(ExternalUrl).Append("\n");
      
      sb.Append("  ExternalUrlSource: ").Append(ExternalUrlSource).Append("\n");
      
      sb.Append("  ExternalUrlType: ").Append(ExternalUrlType).Append("\n");
      
      sb.Append("  ImagePath: ").Append(ImagePath).Append("\n");
      
      sb.Append("  ImageUrl: ").Append(ImageUrl).Append("\n");
      
      sb.Append("  ImageHeight: ").Append(ImageHeight).Append("\n");
      
      sb.Append("  ImageWidth: ").Append(ImageWidth).Append("\n");
      
      sb.Append("  LogoUrl: ").Append(LogoUrl).Append("\n");
      
      sb.Append("  SquareLogoUrl: ").Append(SquareLogoUrl).Append("\n");
      
      sb.Append("  NeedsSanitize: ").Append(NeedsSanitize).Append("\n");
      
      sb.Append("  Summary: ").Append(Summary).Append("\n");
      
      sb.Append("  Topic: ").Append(Topic).Append("\n");
      
      sb.Append("  ApprovalStatus: ").Append(ApprovalStatus).Append("\n");
      
      sb.Append("  ApprovalTransactions: ").Append(ApprovalTransactions).Append("\n");
      
      sb.Append("  GroupIds: ").Append(GroupIds).Append("\n");
      
      sb.Append("  Slug: ").Append(Slug).Append("\n");
      
      sb.Append("  ArticleId: ").Append(ArticleId).Append("\n");
      
      sb.Append("  Comment: ").Append(Comment).Append("\n");
      
      sb.Append("  NewsletterIds: ").Append(NewsletterIds).Append("\n");
      
      sb.Append("  IsFeatured: ").Append(IsFeatured).Append("\n");
      
      sb.Append("  AdvisorId: ").Append(AdvisorId).Append("\n");
      
      sb.Append("  IsPublished: ").Append(IsPublished).Append("\n");
      
      sb.Append("  IsResponsive: ").Append(IsResponsive).Append("\n");
      
      sb.Append("  IsProxyNeeded: ").Append(IsProxyNeeded).Append("\n");
      
      sb.Append("  IsMobileProxyNeeded: ").Append(IsMobileProxyNeeded).Append("\n");
      
      sb.Append("  ProxyUrl: ").Append(ProxyUrl).Append("\n");
      
      sb.Append("  Video: ").Append(Video).Append("\n");
      
      sb.Append("  PdfAttachmentUrl: ").Append(PdfAttachmentUrl).Append("\n");
      
      sb.Append("  PostDate: ").Append(PostDate).Append("\n");
      
      sb.Append("  DisplayDate: ").Append(DisplayDate).Append("\n");
      
      sb.Append("  SuitabilityScore: ").Append(SuitabilityScore).Append("\n");
      
      sb.Append("  VideoId: ").Append(VideoId).Append("\n");
      
      sb.Append("  DisplayTag: ").Append(DisplayTag).Append("\n");
      
      sb.Append("  DisplaySummary: ").Append(DisplaySummary).Append("\n");
      
      sb.Append("  VestorlyUrl: ").Append(VestorlyUrl).Append("\n");
      
      sb.Append("  Title: ").Append(Title).Append("\n");
      
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  
  
}
